wdl-parsing/WDLParsingFunction/src/main/java/io/dockstore/wdlparser/App.java

```diff
@@ -0,0 +1,196 @@
+    strings.remove(0);
+    // If there are no imports, womtool says None
+    if (strings.get(0).equals("None")) {
+      strings.remove(0);
+    }
+    response.setSecondaryFilePaths(strings);
+  }
+
+  @Override
+  public APIGatewayProxyResponseEvent handleRequest(
+      final APIGatewayProxyRequestEvent input, final Context context) {
+
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Content-Type", MediaType.APPLICATION_JSON);
+
+    APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent().withHeaders(headers);
+    if (input != null && input.getBody() != null) {
+      try {
+        LanguageParsingRequest request =
+            mapper.readValue(input.getBody(), LanguageParsingRequest.class);
+        try {
+          String s =
+              parseWdlFile(
+                  request.getUri(),
+                  request.getBranch(),
+                  request.getDescriptorRelativePathInGit(),
+                  request);
+          return response.withStatusCode(HttpURLConnection.HTTP_OK).withBody(s);
+        } catch (IOException e) {
+          String errorMessage = "Could not clone repository to temporary directory";
+          LOGGER.error(errorMessage, e);
+          return response
+              .withBody(errorMessage)
+              .withStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+        } catch (GitAPIException e) {
+          StringWriter sw = new StringWriter();
+          e.printStackTrace(new PrintWriter(sw));
+          String exceptionAsString = sw.toString();
+          return response
+              .withBody(exceptionAsString)
+              .withStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+        }
+      } catch (IOException e) {
+        String errorMessage = "Could not process request";
+        LOGGER.error(errorMessage, e);
+        return response.withBody(errorMessage).withStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+      }
+    } else {
+      return response
+          .withBody("No body in request")
+          .withStatusCode(HttpURLConnection.HTTP_BAD_REQUEST);
+    }
+  }
+
+  private String parseWdlFile(
+      String uri,
+      String branch,
+      String descriptorRelativePathInGit,
+      LanguageParsingRequest languageParsingRequest)
+      throws IOException, GitAPIException {
+    Path tempDirWithPrefix = Files.createTempDirectory("clonedRepository");
+    Git.cloneRepository()
+        .setCloneAllBranches(false)
+        .setBranch(branch)
+        .setURI(uri)
+        .setDirectory(tempDirWithPrefix.toFile())
+        .call();
+    Path descriptorAbsolutePath = tempDirWithPrefix.resolve(descriptorRelativePathInGit);
+    String descriptorAbsolutePathString = descriptorAbsolutePath.toString();
+    LanguageParsingResponse response = getResponse(descriptorAbsolutePathString);
+    response.setLanguageParsingRequest(languageParsingRequest);
+    // Deleting a directory without Common IO
+    Files.walk(tempDirWithPrefix)
```

Comment:
Could go into a finally, although if the clone fails with an exception, there may not be any files in there.